internal/protocol/tcp_server.go

```diff
@@ -26,11 +28,17 @@ func TCPServer(listener net.Listener, handler TCPHandler, logf lg.AppLogFunc) {
 			// theres no direct way to detect this error because it is not exposed
 			if !strings.Contains(err.Error(), "use of closed network connection") {
 				logf(lg.ERROR, "listener.Accept() - %s", err)
+				fatalErr = err
 			}
 			break
 		}
 		go handler.Handle(clientConn)
 	}
 
 	logf(lg.INFO, "TCP: closing %s", listener.Addr())
+	if fatalErr != nil {
+		return errors.New("accept new tcp connection error")
+	}
+
+	return nil
```

Comment:
You could just `return fatalErr` in all cases, it will be `nil` if never set.

But actually, I suggest reverting this function to not return an error at all.